lib/ansible/modules/cloud/azure/azure_rm_monitordiagnosticsettings.py

```diff
@@ -15,36 +15,36 @@
 
 DOCUMENTATION = '''
 ---
-module: azure_rm_diagnosticsettings
+module: azure_rm_monitordiagnosticsettings
 version_added: "2.9"
-short_description: Manage Azure diagnostic setting.
+short_description: Manage Azure monitor diagnostic setting.
 description:
-    - Create, updata and delete an diagnostic setting.
+    - Create, updata and delete an Azure monitor diagnostic setting.
 options:
     name:
         description:
             - The name of the diagnostic setting.
-    target:
+    resource_id:
         description:
             - The target which is the diagnostic settings used for.
             - The identifier of the resource.
+            - The name and the resource_group can be provided as a dict to fill the I(resource_id)
     state:
         default: present
         description:
             -Assert the state of the diagnostic setting. Use C(present) to create or update and C(absent) to delete.
         choices:
             - present
             - absent
-    storage_account_id:
+    storage_account:
         description:
-            - The resource ID of the storage account to which you would like to send Diagnostic Logs.
-    service_bus_rule_id:
+            - The resource name or ID of the storage account to which you would like to send Diagnostic Logs.
+            - The name, resource_group and subscription(optional) can be provided as a dict to fill the I(storage_account)
```

Comment:
It can be a dict contains C(resource_group) and C(name), C(subscription) optional .....